src/Components/Browser.JS/src/Platform/Circuits/AutoReconnectCircuitHandler.ts

```diff
@@ -0,0 +1,33 @@
+import { CircuitHandler } from './CircuitHandler';
+export class AutoReconnectCircuitHandler implements CircuitHandler {
+  modal: HTMLDivElement;
+  message: Text;
+
+  constructor(private maxRetries: number = 5, private retryInterval: number = 3000) {
+    this.modal = document.createElement('div');
+    this.modal.className = 'modal';
+    this.message = document.createTextNode('');
+    this.modal.appendChild(this.message);
+    document.addEventListener('DOMContentLoaded', () => document.body.appendChild(this.modal));
+  }
+  onConnectionUp() {
+    this.modal.style.display = 'none';
+  }
+  async onConnectionDown() {
+    this.message.textContent = 'Attempting to reconnect to the server...';
+
+    this.modal.style.display = 'block';
+    const delay = () => new Promise((resolve) => setTimeout(resolve, this.retryInterval));
+    for (let i = 0; i < this.maxRetries; i++) {
+      await delay();
+      try {
+        await window['Blazor'].reconnect();
+        break;
+      } catch (err) {
+        console.error(err);
+      }
+    }
+
+    this.message.textContent = 'Failed to connect to server.';
```

Comment:
At this point, we give up permanently? I'd say that's a perfectly OK starting point for this feature.

As a future follow-up, we might want to have a `this.modal` contain a button labelled "Retry" that (1) is disabled while line 24 is in progress, (2) immediately ends the delay on line 22 if you click it while we're waiting, (3) restarts the whole `onConnectionDown` method if you click it after we gave up entirely. This is not to suggest you should implement this stuff right now, because we don't even know if it's the right design!